CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -127,50 +130,76 @@ static function int GetSoldierInfiltration(StateObjectReference UnitRef)
 
 	UnitState = XComGameState_Unit(History.GetGameStateForObjectID(UnitRef.ObjectID));
 
-	if(!UnitState.IsSoldier())
-		return default.PERSONNEL_INFIL;
-
 	UnitInfiltration = 0;
 
-	CurrentInventory = UnitState.GetAllInventoryItems();
-	foreach CurrentInventory(InventoryItem)
-	{
-		Template = InfilMgr.GetInfilTemplateFromItem(InventoryItem.GetMyTemplateName());
-
-		if (Template == none)
-			continue;
+	// check character
+	Template = InfilMgr.GetInfilTemplateFromCharacter(UnitState.GetMyTemplate());
```

Comment:
Bit confusing what `Template` refers to, `InfilModTemplate` would be a better name